utilities/simulator_cache/cache_simulator.h

```diff
@@ -38,22 +61,102 @@ class CacheSimulator {
   CacheSimulator& operator=(CacheSimulator&&) = delete;
 
   virtual void Access(const BlockCacheTraceRecord& access);
-  void reset_counter() { sim_cache_->reset_counter(); }
-  double miss_ratio();
-  uint64_t total_accesses();
+  void reset_counter() {
+    num_misses_ = 0;
+    num_accesses_ = 0;
+    user_accesses_ = 0;
+    user_misses_ = 0;
+  }
+  double miss_ratio() {
+    if (num_accesses_ == 0) {
+      return -1;
+    }
+    return static_cast<double>(num_misses_ * 100.0 / num_accesses_);
+  }
+  uint64_t total_accesses() { return num_accesses_; }
+
+  double user_miss_ratio() {
```

Comment:
Make it const?

Reply:
I added const for all functions that do not update member variables.